v4/export/dump.go

```diff
@@ -196,76 +191,237 @@ func (d *Dumper) Dump() (err error) {
 	return nil
 }
 
-func dumpDatabases(pCtx context.Context, conf *Config, connectPool *connectionsPool, writer Writer, rebuildConnFunc func(*sql.Conn) (*sql.Conn, error)) error {
+func (d *Dumper) dumpDatabases(connectPool *connectionsPool, writer *Writer) error {
+	ctx, conf := d.ctx, d.conf
 	allTables := conf.Tables
-	g, ctx := errgroup.WithContext(pCtx)
-	tableDataIRTotal := make([]TableDataIR, 0, len(allTables))
-	splitChunkStart := time.Now()
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithCancel(ctx)
```

Comment:
I think this looks clearer.
```suggestion
	conf := d.conf
	allTables := conf.Tables
	ctx, cancel := context.WithCancel(d.ctx)
```